tailcfg/tailcfg.go

```diff
@@ -2982,3 +2982,31 @@ const LBHeader = "Ts-Lb"
 // correspond to those IPs. Any services that don't correspond to a service
 // this client is hosting can be ignored.
 type ServiceIPMappings map[ServiceName][]netip.Addr
+
+// ClientAuditAction represents an auditable action that a client can report to the
+// control plane.  These actions must correspond to the supported actions
+// in the control plane.
+type ClientAuditAction string
+
+const (
+	// AuditNodeDisconnect action is sent when a node has disconnected
+	// from the control plane.  The details must include a reason in the Details
+	// field, either generated, or entered by the user.
+	AuditNodeDisconnect = ClientAuditAction("DISCONNECT_NODE")
+)
+
+// AuditLogRequest represents an audit log request to be sent to the control plane.
+//
+// This is JSON-encoded and sent over the control plane connection to:
+// POST https://<control-plane>/machine/audit-logs
+type AuditLogRequest struct {
+	// NodeKey is the client's current node key.
+	NodeKey key.NodePublic `json:",omitzero"`
```

Comment:
Normally all machine request types include both NodeKey and the `Version CapabilityVersion` (e.g. SetDNSRequest, SetDeviceAttributesRequest, TokenRequest, ...)